model_eval.py

```diff
@@ -37,6 +38,10 @@ def train_model(self):
             device=self.device,
             max_length = self.max_length
         )
+        
+        if self.task == "dep":
+            dataset_dict, labels_mapper_dep, labels_mapper_head = dataset_prep.run()
+
         dataset_dict, labels_mapper = dataset_prep.run()

```

Comment:
This should be in an else statement, because otherwise it will overrun the 'dataset_dict' you have inside the if above, no?